csharp/private/macros/nuget.bzl

```diff
@@ -0,0 +1,106 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def _get_build_file_content(build_file, build_file_content):
+    if build_file == None and build_file_content == None:
+        return """
+load("@d2l_rules_csharp//csharp:defs.bzl", "setup_basic_nuget_package")
+setup_basic_nuget_package()
+"""
+
+    return build_file_content
+
+def import_nuget_package(
+        name,
+        dir = None,
+        file = None,
+        build_file = None,
+        build_file_content = None,
+        sha256 = None):
+    """Import a vendored NuGet package from either a directory or .nupkg file.
+
+    Rather than downloading the package over HTTP from a package source, an
+    extracted package directory or a .nupkg file can be used. These may be
+    checked into your repository within your workspace, or somewhere else on
+    your disk.
+
+    Exactly one of either the dir or file arguments must be provided. They can
+    either be a relative path in the workspace, or an absolute path. Note that
+    paths on Windows use forward slashes.
+
+    At most one of build_file or build_file_content may be provided. If neither
+    are, a default BUILD file will be used that makes a "best effort" to wire
+    up the package. See docs/UsingNuGetPacakges.md for more info.
+
+    Args:
+      name: A unique name for the package's workspace.
+      dir: A directory containing an extracted package.
```

Comment:
I think it'd be nicer to not have "custom" names here, but I guess we need to disambiguate the directory case?

Reply:
What do you mean by a custom name? Like `file` vs. `dir` as mutually exclusive args?
I didn't see an obvious way to do "is this path a file or folder" in Starlark but I also didn't try hard :)